packages/template-block/src/components/TemplateText.tsx

```diff
@@ -34,33 +45,38 @@ export const TemplateText = ({
 }: TemplateTextProps) => {
     const pageCountStyles = BlockStyles[TextStyles.imageCaption];
     const pageCountLabel = pageCount === 1 ? 'page' : 'pages';
-    const memoTitleRte = useMemo(() => {
-        return (
-            <RichTextEditor
-                id="template-block-title"
-                value={title || convertToRteValue(TextStyles.heading3)}
-                placeholder="Add a title"
-                onTextChange={setTitle}
-                isEditing={isEditing}
-                plugins={getTitlePlugin()}
-            />
-        );
-    }, [title, isEditing, setTitle]);
 
-    const memoDescriptionRte = useMemo(() => {
-        return (
-            <RichTextEditor
-                id="template-block-description"
-                value={description || convertToRteValue()}
-                placeholder={
-                    'Add a description that will be displayed in the block\n\nNote: When template description is available, it will be added by default'
-                }
-                onTextChange={setDescription}
-                isEditing={isEditing}
-                plugins={getDefaultPluginsWithLinkChooser(appBridge)}
-            />
-        );
-    }, [description, isEditing, setDescription, appBridge]);
+    const customTitlePlugins = useMemo(() => {
+        return new PluginComposer()
+            .setPlugin([new SoftBreakPlugin(), new TextStylePlugin({ textStyles: TextStylePluginsWithoutImage })])
+            .setPlugin([new BoldPlugin(), new ItalicPlugin(), new UnderlinePlugin(), new StrikethroughPlugin()])
+            .setPlugin([new ResetFormattingPlugin(), new AutoformatPlugin()]);
+    }, []);
+
+    const memoTitleRte = (
+        <RichTextEditor
+            id="template-block-title"
+            value={title ?? convertToRteValue(TextStyles.heading3)}
+            placeholder="Add a title"
+            onTextChange={setTitle}
```

Comment:
just a note: this will call the API every time onTextChange emits. There is some built-in debounce so it is probably not an issue, but something to be mindful of.

Reply:
Got it. Thanks for noticing 👍